src/hotspot/share/gc/parallel/psCompactionManager.cpp

```diff
@@ -114,42 +118,42 @@ ParCompactionManager::gc_thread_compaction_manager(uint index) {
   return _manager_array[index];
 }
 
-inline void ParCompactionManager::publish_and_drain_oop_tasks() {
-  oop obj;
-  while (oop_stack()->pop_overflow(obj)) {
-    if (!oop_stack()->try_push_to_taskqueue(obj)) {
-      follow_contents(obj);
-    }
-  }
-  while (oop_stack()->pop_local(obj)) {
-    follow_contents(obj);
-  }
+void ParCompactionManager::push_objArray(oop obj) {
+  assert(obj->is_objArray(), "precondition");
+  _mark_and_push_closure.do_klass(obj->klass());
+
+  objArrayOop obj_array = objArrayOop(obj);
+  size_t array_length = obj_array->length();
+  size_t initial_chunk_size =
+    // The destination array is unused when processing states.
```

Comment:
This comment is not very helpful; I had to look into the callee to find out it's referring to `nullptr` in the args and why one can use `nullptr` here, but that process is fairly straightforward, so this comment can probably be dropped.

Reply:
Fixed